test/test_namedtensor.py

```diff
@@ -1824,6 +1824,34 @@ def test_dot(self):
                 args=(create('C:2'), create('W:2')),
                 expected_names=[])
 
+    def test_comparison_ops(self):
+        for device in torch.testing.get_all_device_types():
+            a = torch.randn(3, 3, names=('N', 'C'), device=device)
+            b = torch.randn(3, 3, names=('N', 'C'), device=device)
+            scalar = torch.randn([], device=device)
+
+            self.assertEqual((a == b).names, ['N', 'C'])
```

Comment:
nit: It's possible to write less code by having a list of all of the operations:
```
ops = [lambda a, b: a == b, lambda a, b: a != b, ...]
```
and then running them through a for loop. But that makes it harder to `pdb` into and can be less readable